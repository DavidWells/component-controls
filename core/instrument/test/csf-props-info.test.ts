import path from 'path';
import { loadStoriesTests } from './loadTestFiles';

describe('csf-props-info', () => {
  loadStoriesTests(
    {
      stories: { storeSourceFile: false },
      propsLoaders: [
        { name: '@component-controls/react-docgen-info', test: /\.(js|jsx)$/ },
        {
          name: '@component-controls/react-docgen-typescript-info',
          test: /\.(ts|tsx)$/,
        },
        ,
      ],
      components: {
        storeSourceFile: false,
        resolveFile: (componentName, filePath) => {
          if (filePath.includes('@component-controls/blocks')) {
            const resolved = path.resolve(
              path.dirname(filePath),
              `../../../../ui/blocks/src/${componentName}/${componentName}.tsx`,
            );
            return resolved;
          }
          return filePath;
        },
      },
    },
    ['csf', 'props-info'],
  );
});
