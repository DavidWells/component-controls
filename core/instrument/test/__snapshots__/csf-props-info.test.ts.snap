// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`csf-props-info resolve-node-modules.jsx 1`] = `
Object {
  "components": Object {
    "d71f3883f05f82c67d06210306b47220": Object {
      "from": "@component-controls/blocks",
      "importedName": "ControlsTable",
      "info": Object {
        "description": "Table component to display a story's controls and their editors.
Can adapt to multiple groups of controls, displaying them in their own tabs.",
        "displayName": "ControlsTable",
        "methods": Array [],
        "props": Object {
          "collapsible": Object {
            "description": "if false, will nothave a collapsible frame.",
            "parentName": "BlockContainerProps",
            "type": Object {
              "name": "boolean",
              "raw": "boolean",
            },
          },
          "id": Object {
            "description": "id of the story
optional id to be used for the block
if no id is provided, one will be calculated automatically
from the title.",
            "parentName": "StoryInputProps",
            "type": Object {
              "name": "string",
              "raw": "string",
            },
          },
          "name": Object {
            "description": "alternatively you can use the name of a story to load from an external file",
            "parentName": "StoryInputProps",
            "type": Object {
              "name": "string",
              "raw": "string",
            },
          },
          "sxStyle": Object {
            "description": "theme-ui styling object for Block Box",
            "parentName": "BlockContainerProps",
            "type": Object {
              "name": "SystemStyleObject",
              "raw": "SystemStyleObject",
            },
          },
          "title": Object {
            "description": "optional section title for the block.",
            "parentName": "BlockContainerProps",
            "type": Object {
              "name": "string",
              "raw": "string",
            },
          },
        },
      },
      "name": "ControlsTable",
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/ui/blocks/dist/index.js",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
        "name": "@component-controls/blocks",
      },
      "request": "/Users/atanasster/component-controls/ui/blocks/src/ControlsTable/ControlsTable.tsx",
    },
  },
  "kinds": Object {
    "Storybook/Blocks/ControlsTable": Object {
      "component": "ControlsTable",
      "components": Object {
        "ControlsTable": "d71f3883f05f82c67d06210306b47220",
      },
      "parameters": Object {
        "component": "ControlsTable",
      },
      "repository": Object {
        "browse": "https://github.com/ccontrols/component-controls/tree/master/core/instrument/test/fixtures/csf/props-info/resolve-node-modules.jsx",
        "docs": "https://github.com/ccontrols/component-controls/tree/master#readme",
        "issues": "https://github.com/ccontrols/component-controls/issues",
        "name": "@component-controls/instrument",
      },
      "title": "Storybook/Blocks/ControlsTable",
    },
  },
  "stories": Object {},
  "transformed": "import { ControlsTable } from '@component-controls/blocks';

export default {
  title: 'Storybook/Blocks/ControlsTable',
  parameters: {
    component: ControlsTable,
  },
};
",
}
`;
